clap v3.0.0-rc.4 is going to break the console build

It looks like the clap API has changed again in a way that breaks the way that we dealt with the `clap::Clap` => `clap::Parser` renaming that happened at some point in the past (see #173).

Check out the series of interactions with my shell, written below.

I'm not sure if this represents an outright bug in `clap` itself; I'm guessing not, since we've been relying on beta stuff, and so they're probably within their sem-ver rights to break our usage.



```shell
% git remote -v
origin  git@github.com:tokio-rs/console.git (fetch)
origin  git@github.com:tokio-rs/console.git (push)
% git log -1
commit bbce8e78bab3277b6e533d50df3fb2a7463c3746 (HEAD -> main, origin/main, origin/HEAD)
Author: Eliza Weisman <eliza@buoyant.io>
Date:   Mon Dec 6 12:35:46 2021 -0800

    style: fix misc. clippy lints (#190)
  
    * chore(console): fix large enum variant warning
    * style(console): derive `Default` for `PollStats`
    * chore: fix a big pile of dead-code warnings
  
    Signed-off-by: Eliza Weisman <eliza@buoyant.io>
% git status
On branch main
Your branch is up to date with 'origin/main'.

nothing to commit, working tree clean
% grep -A 1 -B 1 'name = "clap"' Cargo.lock  
[[package]]
name = "clap"
version = "3.0.0-beta.5"
% cargo build
    Finished dev [unoptimized + debuginfo] target(s) in 0.05s
% cargo update -p clap
    Updating crates.io index
    Updating clap v3.0.0-beta.5 -> v3.0.0-rc.4
    Removing clap_derive v3.0.0-beta.5
    Updating os_str_bytes v4.2.0 -> v6.0.0
    Removing proc-macro-error v1.0.4
    Removing proc-macro-error-attr v1.0.4
    Removing unicase v2.6.0
% cargo build
   Compiling tokio-console v0.1.0 (/media/pnkfelix/Rust/TurboWish/issue-189/console/console)
error: cannot find derive macro `Clap` in this scope
[... errors trimmed; see details block ...]
```

<details>
<summary>Click to see complete error output from build</summary>

```
error: cannot find derive macro `Clap` in this scope
 --> console/src/config.rs:8:10
  |
8 | #[derive(Clap, Debug)]
  |          ^^^^
  |
note: `Clap` is imported here, but it is only a trait, without a derive macro
 --> console/src/config.rs:2:22
  |
2 | use clap::{ArgGroup, Parser as Clap, ValueHint};
  |                      ^^^^^^^^^^^^^^
  
error: cannot find attribute `clap` in this scope
 --> console/src/config.rs:9:3
  |
9 | #[clap(
  |   ^^^^
  |
  = note: `clap` is in scope, but it is a crate, not an attribute

error: cannot find attribute `clap` in this scope
  --> console/src/config.rs:20:7
   |
20 |     #[clap(default_value = "http://127.0.0.1:6669", value_hint = ValueHint::Url)]
   |       ^^^^
   |
   = note: `clap` is in scope, but it is a crate, not an attribute

error: cannot find attribute `clap` in this scope
  --> console/src/config.rs:28:7
   |
28 |     #[clap(long = "log", env = "RUST_LOG", default_value = "off")]
   |       ^^^^
   |
   = note: `clap` is in scope, but it is a crate, not an attribute

error: cannot find attribute `clap` in this scope
  --> console/src/config.rs:31:7
   |
31 |     #[clap(flatten)]
   |       ^^^^
   |
   = note: `clap` is in scope, but it is a crate, not an attribute

error: cannot find attribute `clap` in this scope
  --> console/src/config.rs:62:7
   |
62 |     #[clap(long = "retain-for", default_value = "6s")]
   |       ^^^^
   |
   = note: `clap` is in scope, but it is a crate, not an attribute

error: cannot find derive macro `Clap` in this scope
  --> console/src/config.rs:69:10
   |
69 | #[derive(Clap, Debug, Clone)]
   |          ^^^^
   |
note: `Clap` is imported here, but it is only a trait, without a derive macro
  --> console/src/config.rs:2:22
   |
2  | use clap::{ArgGroup, Parser as Clap, ValueHint};
   |                      ^^^^^^^^^^^^^^

error: cannot find attribute `clap` in this scope
  --> console/src/config.rs:70:3
   |
70 | #[clap(group = ArgGroup::new("colors").conflicts_with("no-colors"))]
   |   ^^^^
   |
   = note: `clap` is in scope, but it is a crate, not an attribute

error: cannot find attribute `clap` in this scope
  --> console/src/config.rs:73:7
   |
73 |     #[clap(name = "no-colors", long = "no-colors")]
   |       ^^^^
   |
   = note: `clap` is in scope, but it is a crate, not an attribute

error: cannot find attribute `clap` in this scope
  --> console/src/config.rs:77:7
   |
77 |     #[clap(long = "lang", env = "LANG", default_value = "en_us.UTF-8")]
   |       ^^^^
   |
   = note: `clap` is in scope, but it is a crate, not an attribute

error: cannot find attribute `clap` in this scope
  --> console/src/config.rs:81:7
   |
81 |     #[clap(long = "ascii-only")]
   |       ^^^^
   |
   = note: `clap` is in scope, but it is a crate, not an attribute

error: cannot find attribute `clap` in this scope
  --> console/src/config.rs:87:7
   |
87 |     #[clap(
   |       ^^^^
   |
   = note: `clap` is in scope, but it is a crate, not an attribute

error: cannot find attribute `clap` in this scope
  --> console/src/config.rs:97:7
   |
97 |     #[clap(
   |       ^^^^
   |
   = note: `clap` is in scope, but it is a crate, not an attribute

error: cannot find attribute `clap` in this scope
   --> console/src/config.rs:105:7
    |
105 |     #[clap(flatten)]
    |       ^^^^
    |
    = note: `clap` is in scope, but it is a crate, not an attribute

error: cannot find derive macro `Clap` in this scope
   --> console/src/config.rs:110:10
    |
110 | #[derive(Clap, Debug, Copy, Clone)]
    |          ^^^^
    |
note: `Clap` is imported here, but it is only a trait, without a derive macro
   --> console/src/config.rs:2:22
    |
2   | use clap::{ArgGroup, Parser as Clap, ValueHint};
    |                      ^^^^^^^^^^^^^^

error: cannot find attribute `clap` in this scope
   --> console/src/config.rs:113:7
    |
113 |     #[clap(long = "no-duration-colors", parse(from_flag = std::ops::Not::not), group = "colors")]
    |       ^^^^
    |
    = note: `clap` is in scope, but it is a crate, not an attribute

error: cannot find attribute `clap` in this scope
   --> console/src/config.rs:117:7
    |
117 |     #[clap(long = "no-terminated-colors", parse(from_flag = std::ops::Not::not), group = "colors")]
    |       ^^^^
    |
    = note: `clap` is in scope, but it is a crate, not an attribute

warning: unused imports: `ArgGroup`, `ValueHint`
 --> console/src/config.rs:2:12
  |
2 | use clap::{ArgGroup, Parser as Clap, ValueHint};
  |            ^^^^^^^^                  ^^^^^^^^^
  |
  = note: `#[warn(unused_imports)]` on by default

error[E0599]: no function or associated item named `parse` found for struct `Config` in the current scope
  --> console/src/main.rs:29:36
   |
29 |     let mut args = config::Config::parse();
   |                                    ^^^^^ function or associated item not found in `Config`
   |
  ::: console/src/config.rs:16:1
   |
16 | pub struct Config {
   | ----------------- function or associated item `parse` not found for this
   |
   = help: items from traits can only be used if the trait is implemented and in scope
   = note: the following traits define an item `parse`, perhaps you need to implement one of them:
           candidate #1: `object::read::elf::file::FileHeader`
           candidate #2: `object::read::macho::file::MachHeader`
           candidate #3: `object::read::pe::file::ImageNtHeaders`
           candidate #4: `Parser`
           candidate #5: `nom::internal::Parser`
           candidate #6: `nom::sequence::Tuple`

warning: unused import: `clap::Parser`
 --> console/src/main.rs:5:5
  |
5 | use clap::Parser as Clap;
  |     ^^^^^^^^^^^^

For more information about this error, try `rustc --explain E0599`.
warning: `tokio-console` (bin "tokio-console") generated 2 warnings
error: could not compile `tokio-console` due to 18 previous errors; 2 warnings emitted
% 
```

</details>

--- a/console/Cargo.toml
+++ b/console/Cargo.toml
@@ -8,7 +8,7 @@ repository = "https://github.com/tokio-rs/console"
 [dependencies]
 atty = "0.2"
 console-api = { path = "../console-api", features = ["transport"] }
-clap = "3.0.0-beta.5"
+clap = { version = "3.0.0-beta.5", features = ["cargo", "derive", "env"] }
 tokio = { version = "1", features = ["full", "rt-multi-thread"] }
 tonic = { version = "0.6", features = ["transport"] }
 futures = "0.3"

